design-editor/src/panel/preview/preview-element-vd.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-console */
+/* eslint-disable max-len */
+function startVD (profile) {
+	let stDevice, stProperty, stValue, eventType;
+	const previewEl = document.querySelector('.closet-preview-frame').contentWindow;
+	const tau = previewEl.tau;
+
+	/* iFrame Element */
+	const textData = previewEl.document.querySelector('input[type=\'text\']');
+	const sliderData = previewEl.document.querySelector('input[type=\'range\']');
+	const dimmerData = previewEl.document.querySelector('.ui-dimmer');
+	const graphData = previewEl.document.querySelector('.ui-graph');
+	/* TAU widget Element */
+	const dimmer = tau.widget.Dimmer(dimmerData);
+	const graph = tau.widget.Graph(graphData);
+	const slider = tau.widget.Slider(sliderData);
+	const text = tau.widget.TextInput(textData);
+
+	const connection = new WebSocket('ws://localhost:8000');
+
+	connection.onopen = function () {
+		console.log('WebSocket is connected.');
+	};
+
+	connection.onerror = function (error) {
+		console.log(`not connect or server is down, ${error}`);
+	};
+
+	connection.onmessage = function (message) {
+		const json = JSON.parse(message.data);
+		const ElementArray = [];
+		if (json.type === 'oic.r.light.dimming') {
+			stDevice = json.type;
+			stProperty = json.data.text;
+			stValue = json.data.id;
+			console.log(`stDevice : ${stDevice} / stProperty : ${stProperty} / stValue : ${stValue}`);
+
+			if (textData != undefined || textData != null) {
+				ElementArray.push(textData);
+			}
+			if (sliderData != undefined || sliderData != null) {
+				ElementArray.push(sliderData);
+			}
+			if (dimmerData != undefined || dimmerData != null) {
+				ElementArray.push(dimmerData);
+			}
+			if (graphData != undefined || graphData != null) {
+				ElementArray.push(graphData);
+			}
+			compareAttribute(ElementArray);
+		} else {
+			console.log(`Incorrect data : ${json}`);
+		}
+	};
+
+	if (profile === 'mobile') {
+		eventType = 'input';
+	} else {
+		eventType = 'change';
+	}
+
+	if (sliderData != undefined || sliderData != null) {
+		sliderData.addEventListener(eventType, (event) => {
+			const data = event.target.value;
+			const dataInfo = {
+				device: 'oic.r.light.dimming',
+				property: 'dimmingSetting',
+				value: data
+			};
+			if (textData != undefined || textData != null) {
```

Comment:
Why on change slider, we always update textData? Are they connected somehow?

Reply:
Yes. There is a Slider or Text to confirm the change in the dimmer.
Confirm the change if a Slider or Text or both are added.